### ISSUE TITLE
Can I use HTML in the Badge text field in the admin panel?

### ISSUE THREAD

**Issue by muhyal** [CharlieEtienne/sold-out-badge-for-woocommerce#18]
Hi,

Can I use HTML in the Badge text field in the admin panel? 

If I use HTML it shows the entered value as plain text:

![2022-09-07_20-07-32](https://user-images.githubusercontent.com/55546452/188938368-c3ed83ce-5bd8-4590-a8a1-0ef728d1e847.jpg)

For example, for a badge that looks like this:

![2022-09-07_20-06-16](https://user-images.githubusercontent.com/55546452/188938411-625e006f-0244-4ce8-9083-7eaa94ed8dc9.jpg)

I want to set it like this:

![2022-09-07_20-06-22](https://user-images.githubusercontent.com/55546452/188938447-3ef2ecf1-c38f-44c4-ab52-fdb3d82d21e3.jpg)

I did some research and found a document like this, but I couldn't get a successful result:
https://carbonfields.net/docs/fields-html/

Thank you for your support.

**Comment by CharlieEtienne** [comment]
Hi @muhyal, 

In fact this is not a limitation of carbon fields or my plugin, but a limitation from WooCommerce. The hook used to display the badge is escaping html.

So the workarounds would be to override the products pages in your theme, or to replace WC hook.

**Comment by CharlieEtienne** [comment]
Wait, I'm not sure of what I told in my previous message... Let me check.

**Comment by CharlieEtienne** [comment]
I was completely wrong, sorry. I was thinking to another plugin. The HTML is actually escaped in my plugin, in `wp-content/plugins/sold-out-badge-for-woocommerce/woocommerce/single-product/sold-out.php` on line 15.

Could you try this code (in your theme's `functions.php` for example) and let me know if it works?

```php
global $post, $product;
add_filter('wcsob_soldout', function($content) {
    return htmlspecialchars_decode($content, ENT_QUOTES);
}, $post, $product);
```

**Comment by muhyal** [comment]
I tested it but this time the theme is broken.

It looks like this:

![2022-09-08_11-07-19](https://user-images.githubusercontent.com/55546452/189069977-f98b30d6-38e2-4bb6-b8cc-8fb96e956a54.jpg)

**Comment by CharlieEtienne** [comment]
Yes, I forgot to specify the number of arguments. This is the right function to use:

```php
global $post, $product;
add_filter('wcsob_soldout', function ($content, $post, $product) {
	return htmlspecialchars_decode($content, ENT_QUOTES);
}, 10, 3);
```

**Comment by muhyal** [comment]
You've probably heard this a lot in your life @CharlieEtienne . "You are a great person" :) Thanks for your support.

Looks just like we imagined:

![2022-09-08_15-56-31](https://user-images.githubusercontent.com/55546452/189127435-14b73c80-6437-4714-94ed-8893ff8c0523.jpg)

**Comment by CharlieEtienne** [comment]
You're welcome!
